File: Lua_5.4/1047 - Tempo de Jogo com Minutos.lua
--- author: Josélio de S. C. Júnior <joseliojrx25@gmail.com>
--- copyright: Josélio de S. C. Júnior 2022

local ih, im, eh, em = io.read 'n', io.read 'n', io.read 'n', io.read 'n'
local function p(h, m) print(('O JOGO DUROU %d HORA(S) E %d MINUTO(S)'):format(h, m)) end
local every = (ih == eh and im == em) and true
local tm = (ih * 60 + im) - (eh * 60 + em)
if every then
  p(24, 0)
elseif ih == eh then
  local mm = im - em
  if mm > 0 then p(23, 60 - mm)
  else p(0, mm * -1) end
else
  local m = 60 - (tm % 60)
  if tm > 0 then p(math.floor((1440 - tm) / 60), m == 60 and 0 or m)
  else p(math.ceil(tm / 60) * -1, m == 60 and 0 or m) end
end

File: C#_mono_5.10.1.20/1047 - Tempo de Jogo com Minutos.cs
/// author Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
/// copyright Josélio Júnior (Lunatic Fox) 2022

using System;
using System.Linq;

class URI {
  static void Main(string[] args) { 
    int[] r = (Console.ReadLine() ?? "")
      .Split(' ')
      .Select(e => int.Parse(e))
      .ToArray();

    int ih = r[0],
      im = r[1],
      eh = r[2],
      em = r[3];

    if (ih == eh && im == em) {
      Console.WriteLine(P(24, 0));
    } else if (ih == eh) {
      int mm = im - em;
      Console.WriteLine(mm > 0 ? P(23, 60 - mm) : P(0, mm * -1));
    } else {
      int tm = (ih * 60 + im) - (eh * 60 + em);
      int m = tm > 0 ? 60 - (tm % 60) : tm % 60 * -1;

      Console.WriteLine(
        tm > 0 ? 
        P(Convert.ToInt32(Math.Floor((1440 - tm) / 60.0)), m == 60 ? 0 : m)
        : P(Convert.ToInt32(Math.Ceiling(tm / 60.0)) * -1, m == 60 ? 0 : m)
      );
    }
  }
    
  static string P(int h, int m) {
      return $"O JOGO DUROU {h} HORA(S) E {m} MINUTO(S)";
  }
}
